homeassistant/helpers/script.py

```diff
@@ -68,9 +69,21 @@ def script_delay(now):
                         self._delay_listener = None
                         self.run(variables)
 
+                    delay = action[CONF_DELAY]
+
+                    if isinstance(delay, str):
+                        time_template = datetime.strptime(
```

Comment:
Instead of using `strptime`, pass it through the original validation to convert it to a timedelta.

```python
delay = vol.All(cv.time_period, cv.positive_timedelta)(template.render(self.hass, delay)
```